polkadot/node/core/pvf/src/artifacts.rs

```diff
@@ -88,37 +128,28 @@ impl ArtifactId {
 		Self::new(pvf.code_hash(), pvf.executor_params().hash())
 	}
 
-	/// Tries to recover the artifact id from the given file name.
-	#[cfg(test)]
-	pub fn from_file_name(file_name: &str) -> Option<Self> {
-		use polkadot_core_primitives::Hash;
-		use std::str::FromStr as _;
-
+	/// Returns the expected path to this artifact given the root of the cache.
+	pub fn path_prefix(&self, cache_path: &Path) -> PathBuf {
```

Comment:
`path_prefix` is not really an accurate name, it's more of a filename prefix. Maybe `path_with_prefix`. But I wonder whether we still need this fn at all.

Reply:
Hmm, not really a filename prefix. This is the prefix of the expected path, without the checksum, which used to be the canonical path of an artifact.

Reply:
Idk, `path_with_prefix` conveys that the return is a path, but it is more of a prefix of the canonical path.

Reply:
Now that I think of it, maybe we should just do a `path(cache_path, checksum)` instead, which is probably more useful.